### PR TITLE
feat: Add deactivate only mode

### LINKED ISSUE #38: Deactivate only mode
### Description
We got a request, if we can implement a "deactivate only" mode. In this mode, it will only check for deactivated users and will not provision new users.

### User story
As a Famedly customer, I want to use ldap-sync for the sole purpose of deactivating users in Famedly that have already been deactivated in my LDAP.

### Steps to do
- Add a feature flag
- Only search for deactivated users per the configured `disable_bitmasks`.
- Find matching users in Zitadel
- Disable matching users

### PR DESCRIPTION
**Pull request by mzaniolo**
Closes #38

### CODE REVIEW

**Review by sirewix — APPROVED** [review]

File: src/lib.rs [hunk 1]
```diff
@@ -48,9 +50,12 @@ pub async fn sync_ldap_users_to_zitadel(config: Config) -> Result<()> {
 
 	sync_handle.await??;
 
-	zitadel.import_new_users(added).await?;
 	zitadel.update_users(changed).await?;
```
Comment (emgrav) [inline src/lib.rs:53]: Won't this still mean that a disabled user being enabled will also propagate from LDAP to Zitadel?

File: src/config.rs [hunk 2]
```diff
@@ -279,6 +279,8 @@ pub enum FeatureFlag {
 	VerifyPhone,
 	/// If set, only log changes instead of writing anything
 	DryRun,
+	///If only deactivated user should be sync
```
Comment (emgrav) [inline src/config.rs:282]: Just a tiny nit
```suggestion
	/// If only deactivated users should be synced
```

**Review by emgrav — CHANGES_REQUESTED** [review]
I have one question about the functionality and a tiny nit about a comment, but looks quite good otherwise.

File: src/lib.rs (hunk 1 above)
Comment (mzaniolo) [reply src/lib.rs:53]: No, inside zitadel.rs we have a differentiation between the changes to enable and disable (line 91). I have another check for this feature flag on the zitadel.rs to handle this case. However, I just extended my test case to also cover this scenario

**Review by emgrav — APPROVED** [review]
LGTM

File: src/lib.rs [hunk 3]
```diff
@@ -24,6 +24,8 @@ pub async fn sync_ldap_users_to_zitadel(config: Config) -> Result<()> {
 	let zitadel = Zitadel::new(&config).await?;
 	let (mut ldap_client, ldap_receiver) = Ldap::new(config.clone().ldap.into(), cache);
 
+	let deactivate_only = config.feature_flags.contains(&FeatureFlag::DeactivateOnly);
```
Comment (tlater-famedly) [inline src/lib.rs:27]: Small nit: We could have a `config.disable_only()` so we don't need to do this `.contains` shenanigans all the time.

Comment (sirewix) [reply src/lib.rs:27]: We do have such function for some other feature flag. I'd say we should add some generic method to config:
```rust
pub fn is_feature_enabled(feature_flag: FeatureFlag) -> bool {...}
```
However the ".contains shenanigans" are quite readable, so whatever, but let's make it consistent with other "is feature X enabled" expressions, explicit or hidden behind some function

**Comment by sirewix** [comment]
Can you add some lines to the readme and/or config sample on how cache behaves when this mode is enabled. From what I see the cache is written as if all changes were processed, so it wouldn't be possible to run with `disable_only` and then without and get every change processed that wasn't processed during `disable_only` run.

**Comment by mzaniolo** [comment]
> Can you add some lines to the readme and/or config sample on how cache behaves when this mode is enabled. From what I see the cache is written as if all changes were processed, so it wouldn't be possible to run with `disable_only` and then without and get every change processed that wasn't processed during `disable_only` run.

Well, I think we should first be ask how we want the cache to behave when deactivate only is set. The issue doesn't say anything about it. Does anyone know how to answer that? Maybe @nikzen ?

**Review by sirewix — APPROVED** [review]

**Pull request merged**